include/mbedtls/gcm.h

```diff
@@ -146,9 +145,9 @@ int mbedtls_gcm_setkey( mbedtls_gcm_context *ctx,
  * \return          \c 0 if the encryption or decryption was performed
  *                  successfully. Note that in #MBEDTLS_GCM_DECRYPT mode,
  *                  this does not indicate that the data is authentic.
- * \return          #MBEDTLS_ERR_GCM_BAD_INPUT if the lengths are not valid.
- * \return          #MBEDTLS_ERR_GCM_HW_ACCEL_FAILED or a cipher-specific
- *                  error code if the encryption or decryption failed.
+ * \return          #MBEDTLS_ERR_GCM_BAD_INPUT if the lengths are not valid or
+ *                  a cipher-specific error code if the encryption
```

Comment:
A platform error code is not a cipher-specific error code. What makes sense to say here? Should we list `MBEDTLS_ERR_PLATFORM_HW_FAILED` explicitly or say "a cipher- or platform-specific error code"? Same goes for other similar comments for other functions and files.

Reply:
Correct, but the `MBEDTLS_ERR_PLATFORM_HW_FAILED` is a generic error that is returned by the platform. The cipher specific code is because `gcm` is part of the cipher module.

Reply:
Where in this function's documentation do we indicate that you might receive a platform error code? Or is that universally implied or expressly written elsewhere?

Reply:
It should be universally implied. It will be updated in the relevant KB article, about implementing alternative modules

Reply:
OK